src/apps/watchfaces/watchface.cpp

```diff
@@ -40,7 +40,14 @@ void OswAppWatchface::drawStepHistory(OswUI* ui, uint8_t x, uint8_t y, uint8_t w
     hal->gfx()->setTextBottomAligned();
     hal->gfx()->setTextSize(1);
     hal->gfx()->setTextCursor(DISP_W / 2, y - 1);
-    hal->gfx()->print(hal->environment->getStepsToday());
+
+    if (OswConfigAllKeys::settingDisplay.get()) {
+      char stepsTodayNGoal[20];
```

Comment:
This part has been temporarily assigned an appropriate size. 
Memory will be free, but let me know if there's a better way?

Reply:
Sure! As we use the ExpressIf framework (which is based (?) on Arduino, which is based on FreeRTOS) you could use the capabilities of the `String`-class to format it without using a temporary character buffer. I would recommend you to use that instead :smile_cat: